iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c

```diff
@@ -216,13 +212,13 @@ void test_invoke_device_method(const char* deviceId, const char* moduleId, const
     {
         LogInfo("IoTHubDeviceMethod_InvokeModule deviceId='%s', moduleId='%s'", deviceId, moduleId);
         IOTHUB_DEVICE_METHOD_RESULT invokeResult = IoTHubDeviceMethod_InvokeModule(serviceClientDeviceMethodHandle, deviceId, moduleId, METHOD_NAME, payload, TIMEOUT, &responseStatus, &responsePayload, &responsePayloadSize);
-        ASSERT_ARE_EQUAL(IOTHUB_DEVICE_METHOD_RESULT, IOTHUB_DEVICE_METHOD_OK, invokeResult, "Service Client IoTHubDeviceMethod_InvokeModule failed");
+        ASSERT_ARE_EQUAL(int, IOTHUB_DEVICE_METHOD_OK, invokeResult, "Service Client IoTHubDeviceMethod_InvokeModule failed");
```

Comment:
Just curious. How useful are these additional error message strings? For ulib we don't have them and if a test fails it will say something like "Expected X but was Y on line 215". That seems to be sufficient to see where the problem is. These error messages only add to the strings that are required to be stored by the device and potentially further limit platforms that could run them. I could be nitpicking here.

Reply:
I kind of like them because there's been times I've seen the message coming out of the log and I can go right to problem in code without needing to stare at test.

Now that said I can see how putting the e2e on the device itself is already getting tight.  For future projects I'm fine with us having guidance NOT to do this.

---
In reply to: [350351849](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r350351849) [](ancestors = 350351849)